streamlit_pianoroll/frontend/src/pianoroll.ts

```diff
@@ -183,6 +183,9 @@ class PianoRoll {
   }
 
   private updateCurrentPageIdx(current_time: number): void {
+    // Block updating page index when current time is equal total duration as it is the end
+    if (current_time >= this.duration_total) return
```

Comment:
Do we know the reason why this happens? I think we discussed this in proll tasks, but I'm not sure what the conclusion was - is the player sending updates with `current_time` value higher than the `.end` property of the last note present in the piano roll? 

Can you add a short comment explaining how this can happen, and why `duration_total` can be lower than the max `current_time? :)

Reply:
Streamlit didn't have as much difference in times as pianoroll, but as far as I can see it's enough that when `current_time` equals `this.end`, dividing `current_time / this.page_duration` returns a value that exceeds the max number of pages. So instead of checking
```js
if (current_time >= this.duration_total) return
```
We can even modify the calculation of page_index:
```js
const page_index = Math.floor(
  Math.min(current_time / this.page_duration, this.note_pages.length - 1)
)
```

And it should also work :)
I have added a comment about this to the code as you asked

Reply:
Thanks, this is fine, my main point here was to prevent confusion in future readers :)